packages/payment-plugin/src/mollie/mollie.handler.ts

```diff
@@ -0,0 +1,123 @@
+import createMollieClient, { RefundStatus } from '@mollie/api-client';
+import { LanguageCode } from '@vendure/common/lib/generated-types';
+import {
+    CreatePaymentResult,
+    CreateRefundResult,
+    Logger,
+    PaymentMethodHandler,
+    PaymentMethodService,
+    SettlePaymentResult,
+} from '@vendure/core';
+
+import { loggerCtx, PLUGIN_INIT_OPTIONS } from './constants';
+import { MolliePluginOptions } from './mollie.plugin';
+
+let paymentMethodService: PaymentMethodService;
+let options: MolliePluginOptions;
+export const molliePaymentHandler = new PaymentMethodHandler({
+    code: 'mollie-payment-handler',
+    description: [
+        {
+            languageCode: LanguageCode.en,
+            value: 'Mollie payment',
+        },
+    ],
+    args: {
+        apiKey: {
+            type: 'string',
+        },
+        redirectUrl: {
+            type: 'string',
+        },
+    },
+    init(injector) {
+        paymentMethodService = injector.get(PaymentMethodService);
+        options = injector.get(PLUGIN_INIT_OPTIONS);
+    },
+    createPayment: async (ctx, order, amount, args, _metadata): Promise<CreatePaymentResult> => {
```

Comment:
Return type should also include `CreatePaymentErrorResult`, so that in the `catch` block you can return a result with the state `Error`, which is more correct than `Declined`